src/rsz/src/GateCloner.cc

```diff
@@ -0,0 +1,763 @@
+      closest = cand;
+    }
+  }
+  return closest;
+}
+
+std::vector<const Pin *>
+GateCloner::fanoutPins(Net* pin_net, bool include_top_level) const
+{
+  std::vector<const Pin *> filtered_inst_pins;
+  auto inst_pins = resizer_->getPins(pin_net);
+  filtered_inst_pins = filterPins(inst_pins, sta::PortDirection::input(),
+                                  include_top_level);
+
+  if (include_top_level) {
+    std::vector<const Pin *> filtered_top_pins;
+    auto itr = network_->connectedPinIterator(pin_net);
+    while (itr->hasNext()) {
+      const Pin * term = itr->next();
+      if (network_->isTopLevelPort(term) &&
+          network_->direction(term)->isOutput()) {
+        filtered_top_pins.push_back(term);
+      }
+    }
+    delete itr;
+    filtered_inst_pins.insert(filtered_inst_pins.end(),
+                              filtered_top_pins.begin(),
+                              filtered_top_pins.end());
+  }
+  return filtered_inst_pins;
+}
+
+bool
+GateCloner::violatesMaximumCapacitance(Pin* term, float limit_scale_factor)
+{
+  const sta::Corner*   corner;
+  const sta::RiseFall* rf;
+  float                cap, limit, ignore;
+
+  if (!capacitance_limits_initialized_) {
+    sta_->checkCapacitanceLimitPreamble();
+    capacitance_limits_initialized_ = true;
+  }
+
+  // Needed?:  float load_cap = network_->graphDelayCalc()->loadCap(term, resizer_->tgt_slew_dcalc_ap_);
+  sta_->checkCapacitance(term, nullptr, sta::MinMax::max(), corner, rf,
+                         cap, limit, ignore);
+  float diff = (limit_scale_factor * limit) - cap;
+  return diff < 0.0 && limit > 0.0;
+}
+
+bool
+GateCloner::violatesMaximumTransition(Pin* term, float limit_scale_factor)
+{
+  const sta::Corner*   corner;
+  const sta::RiseFall* rf;
+  float                slew, limit, ignore;
+  sta_->checkSlew(term, nullptr, sta::MinMax::max(), false, corner, rf,
+                  slew, limit, ignore);
+  float diff = (limit_scale_factor * limit) - slew;
+  return diff > 0.0 && limit > 0.0;
+}
+
+Vertex* GateCloner::vertex(Pin* term) const
+{
+  Vertex *vertex, *bidirect_drvr_vertex;
+  sta_->graph()->pinVertices(term, vertex, bidirect_drvr_vertex);
+  return vertex;
+}
+
+float GateCloner::pin_slew(Pin* term) const
+{
+  return std::max(slew(term, true), slew(term, false));
+}
+
+float GateCloner::slew(Pin* term, bool is_rise) const
+{
+  return sta_->vertexSlew(vertex(term),
+      is_rise ? sta::RiseFall::rise() : sta::RiseFall::fall(),
+      sta::MinMax::max());
+}
+
+void GateCloner::cloneTree(Instance* inst, float cap_factor,
+                           bool clone_largest_only)
+{
+  std::unique_ptr<InstancePinIterator> inst_pin_iter{network_->pinIterator(inst)};
+  Pin *output_pin = nullptr;
+
+  while (inst_pin_iter->hasNext()) {
+    output_pin = inst_pin_iter->next();
+    if (network_->direction(output_pin)->isOutput()) {
+        break;
+    }
+  }
+  if (output_pin == nullptr || network_->direction(output_pin)->isInput()) {
+  return;
+  }
```

Comment:
We already discussed the problems turning on the current global clang-format would cause. I am fine if we can have a slightly different clang-format file for rsz ... otherwise lets defer this topic for now.